invenio_oauthclient/handlers/token.py

```diff
@@ -112,10 +112,31 @@ def oauth2_token_setter(remote, resp, token_type="", extra_data=None):
         secret="",
         token_type=token_type,
         extra_data=extra_data,
+        refresh_token=resp.get("refresh_token"),
+        expires_at=make_expiration_time(resp.get("expires_in")),
     )
 
 
-def token_setter(remote, token, secret="", token_type="", extra_data=None, user=None):
+def make_expiration_time(expires_in):
+    """Make expiration time from expires_in.
+
+    :param expires_in: The time in seconds.
+    """
+    if expires_in is None:
+        return None
+    return datetime.datetime.now() + datetime.timedelta(seconds=expires_in)
```

Comment:
Question: Why not use `datetime.now(timezone.utc)` instead of `datetime.now()` to ensure all datetime objects are timezone-aware for better [consistency](https://github.com/search?q=repo%3Ainveniosoftware%2Finvenio-oauthclient+now%28&type=code) and clarity?
(`utcnow()` will be deprecated in 3.12)

Reply:
Hmm, if you changed `sa.Column("expires_at", sa.DateTime(timezone=True), nullable=True)` and SQLAlchemy still removes the timezone information despite the change, the problem could be relying somewhere else right? 
WDYT @slint?

Reply:
Pushed the "aware" version just to see the test results for different databases.

It seems that both sqlite and mysql (https://dev.mysql.com/doc/refman/8.0/en/datetime.html) convert timestamp to utc before storing to db and then to local timezone upon retrieval - but "awareness" is lost in sqlalchemy.
Only postgres stores the timezone with the datetime ("timestamp with time zone" type) and subsequently sqlalchemy returns timezone aware datetime.

As invenio sources use utcnow() in many places (grep gave me more that 100) and even the maintrunk of sqlalchemy_utils uses utcnow(), I'd vote for keeping the datetime naive but in utc, using utcnow in the code. Later on when the invenio codebase will be upgraded to python 3.12++, then if not already solved by sqlalchemy itself we might introduce our own timestamp decorator in invenio-db (similar to https://mike.depalatis.net/blog/sqlalchemy-timestamps.html)

@Samk13 @slint What do you think?

Reply:
Switching to `datetime.now(timezone.utc)` for consistency and future-proofing is logical to me. If `sa.DateTime(timezone=True)` strips timezone info, the issue is deeper. 
I would vote to stick with `utcnow()` for now and address this in Python 3.12 upgrades. 
@slint, I recommend this. Your call. Thoughts?